### ISSUE TITLE
[UIImage+Compare] compareWithImage method not available

### ISSUE THREAD

**Issue by ghost** [facebook/react-native#2539]
I'm attempting to create a snapshot test equivalent to SimpleSnapshotTest (in IntegrationTests.m) from the UIExplorer example.

I've been able to run the test with `recordMode = YES` and it created a snapshot image.  However, when I run the test with `recordMode = NO` it throws in FBSnapshotTestController compareReferenceImage with `'NSInvalidArgumentException', reason: '-[UIImage compareWithImage:]: unrecognized selector sent to instance 0x78e6f060'`

I'm brand new to Objective-C so it took me a little bit to figure out that this meant the method wasn't available.  Unfortunately I couldn't see anything missing in the project build configuration compared to the example project.  I have node_modules/react-native/Libraries/RCTTest/RCTTest.xcodeproj added under Libraries in XCode and RCTTest.xcodeproj/Products/libRCTTest.a has been added to "Link Binary With Libraries" in the test target.

The odd thing is that when I tried the suggestion from https://stackoverflow.com/questions/4137988/categories-in-objective-c-arent-working#comment-4473686
it just started working.

How can I fix this?

**TL;DR** UIImage+Compare only gets linked when a random C function gets added to it and called in FBSnapshotTestController.

**Comment by brentvatne** [comment]
cc @tadeuzagallo

**Comment by besarthoxhaj** [comment]
@brentvatne @tadeuzagallo @willatthewindow got the same identical issue

**Comment by machard** [comment]
Same here

**Comment by christopherdro** [comment]
@facebook-github-bot answered
